core/src/sumeragi/mod.rs

```diff
@@ -377,12 +377,18 @@ impl<G: GenesisNetworkTrait, K: KuraTrait, W: WorldTrait> ContextHandler<Init>
 }
 
 #[async_trait::async_trait]
-impl<G: GenesisNetworkTrait, K: KuraTrait, W: WorldTrait> Handler<GetSortedPeers>
+impl<G: GenesisNetworkTrait, K: KuraTrait, W: WorldTrait> Handler<GetRandomPeer>
     for Sumeragi<G, K, W>
 {
-    type Result = Vec<PeerId>;
-    async fn handle(&mut self, GetSortedPeers: GetSortedPeers) -> Vec<PeerId> {
-        self.topology.sorted_peers().to_vec()
+    type Result = PeerId;
+    async fn handle(&mut self, GetRandomPeer: GetRandomPeer) -> Self::Result {
+        let rng = &mut rand::thread_rng();
+        #[allow(clippy::expect_used)]
+        self.topology
+            .sorted_peers()
+            .choose(rng)
+            .expect("No nodes in topology")
+            .clone()
     }
 }
```

Comment:
I think Sumeragi should not concern itself with implementation specifics of block sync, so getting all sorted peers - which is a general case and then in block sync choosing the peer makes more sense to me.

Also I think we can get peers from wsv and it is also always up to date. I think it's better to get it from there as a single central point for other components.

Reply:
you are correct, I missed that there is a method in `wsv`. I'll use that one. Would you like me to return the `GetPeers` message? it's not used anymore

Reply:
Ok, then it is better to remove it I think.

Reply:
done